pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AccessorMethodGenerationRule.java

```diff
@@ -39,34 +38,30 @@ public void end(RuleContext ctx) {
 
     @Override
     public Object visit(ASTFieldAccess node, Object data) {
-        JFieldSymbol sym = node.getReferencedSym();
-        if (sym != null && sym.getConstValue() == null) {
-            checkMemberAccess((RuleContext) data, node, sym);
-        }
+        checkMemberAccessIfConstValueIsNull(node, (RuleContext) data, node.getReferencedSym());
         return null;
     }
 
     @Override
     public Object visit(ASTVariableAccess node, Object data) {
-        JVariableSymbol sym = node.getReferencedSym();
-        if (sym instanceof JFieldSymbol) {
-            JFieldSymbol fieldSym = (JFieldSymbol) sym;
-            if (((JFieldSymbol) sym).getConstValue() == null) {
-                checkMemberAccess((RuleContext) data, node, fieldSym);
-            }
-        }
+        checkMemberAccessIfConstValueIsNull(node, (RuleContext) data, node.getReferencedSym());
         return null;
     }
 
+    private void checkMemberAccessIfConstValueIsNull(JavaNode node, RuleContext data, JVariableSymbol sym) {
+        if (sym instanceof JFieldSymbol && ((JFieldSymbol) sym).getConstValue() == null) {
+            checkMemberAccess(data, (ASTExpression) node, (JFieldSymbol) sym);
+        }
+    }
+
     @Override
     public Object visit(ASTMethodCall node, Object data) {
-        JMethodSymbol symbol = (JMethodSymbol) node.getMethodType().getSymbol();
```

Comment:
no need for casting this way and onetime usages always feel to me like: `(RSPEC-S1488): Local variables should not be declared and then immediately returned or thrown`